src/ui/components/LockModal/LockModal.tsx

```diff
@@ -0,0 +1,183 @@
+import {
+  IonModal,
+  useIonViewDidEnter,
+  useIonViewWillEnter,
+} from "@ionic/react";
+import { i18n } from "../../../i18n";
+import "./LockModal.scss";
+import { RoutePath } from "../../../routes";
+import { PasscodeModule } from "../PasscodeModule";
+import { ErrorMessage, MESSAGE_MILLISECONDS } from "../ErrorMessage";
+import { PageFooter } from "../PageFooter";
+import { Alert } from "../Alert";
+import { ResponsivePageLayout } from "../layout/ResponsivePageLayout";
+import { useAppIonRouter } from "../../hooks";
+import { useAppDispatch, useAppSelector } from "../../../store/hooks";
+import {
+  getAuthentication,
+  login,
+  setAuthentication,
+  setCurrentRoute,
+  setPauseQueueIncomingRequest,
+} from "../../../store/reducers/stateCache";
+import { useEffect, useState } from "react";
+import { KeyStoreKeys, SecureStorage } from "../../../core/storage";
+import { PublicRoutes, TabsRoutePath } from "../../../routes/paths";
+import { LockModalTypes } from "./LockModal.types";
+
+const LockModal = ({ didEnter }: LockModalTypes) => {
+  const componentId = "lock-modal";
+  const ionRouter = useAppIonRouter();
+  const dispatch = useAppDispatch();
+  const authentication = useAppSelector(getAuthentication);
+  const [passcode, setPasscode] = useState("");
+  const seedPhrase = authentication.seedPhraseIsSet;
+  const [alertIsOpen, setAlertIsOpen] = useState(false);
+  const [showModalAfterRender, setShowModalAfterRender] = useState(false);
+  const [passcodeIncorrect, setPasscodeIncorrect] = useState(false);
+  const headerText = seedPhrase
+    ? i18n.t("lockmodal.alert.text.verify")
+    : i18n.t("lockmodal.alert.text.restart");
+  const confirmButtonText = seedPhrase
+    ? i18n.t("lockmodal.alert.button.verify")
+    : i18n.t("lockmodal.alert.button.restart");
+  const cancelButtonText = i18n.t("lockmodal.alert.button.cancel");
+
+  useEffect(() => {
+    didEnter && didEnter();
+    setShowModalAfterRender(true);
+  }, []);
+
+  const handleClearState = () => {
+    setAlertIsOpen(false);
+    setPasscodeIncorrect(false);
+    setPasscode("");
+  };
+
+  useEffect(() => {
+    if (passcodeIncorrect) {
+      setTimeout(() => {
+        setPasscodeIncorrect(false);
+      }, MESSAGE_MILLISECONDS);
+    }
+  }, [passcodeIncorrect]);
+
+  const handlePinChange = (digit: number) => {
+    const updatedPasscode = `${passcode}${digit}`;
+
+    if (updatedPasscode.length <= 6) setPasscode(updatedPasscode);
+
+    if (updatedPasscode.length === 6) {
+      verifyPasscode(updatedPasscode).then((verified) => {
+        if (verified) {
+          dispatch(login());
+          handleClearState();
+          setTimeout(() => {
+            dispatch(setPauseQueueIncomingRequest(false));
```

Comment:
I don't see anything pausing the queue when the app is locked with `setPauseQueueIncomingRequest(true)` unless I'm missing it

Reply:
we are already have `setPauseQueueIncomingRequest(true)` inside the `login` method.
I just removed the timeout.

```
login: (state) => {
      state.authentication = {
        ...state.authentication,
        loggedIn: true,
      };
      state.queueIncomingRequest = {
        ...state.queueIncomingRequest,
        isPaused: false,
        isProcessing: true,
      };
    }
```

Reply:
Ah yes, true. There is a `setPauseQueueIncomingRequest(true)` in the `initApp` - is there a chance this will ever conflict?

Maybe its initial state is just paused and we only unpause it when we login, OR, if we finish onboarding?

Reply:
actually, why do we need to pause the incoming requests in the `initApp`?
I don't see them being reactivated after loading.

Reply:
I'm not sure, it might have been legacy code. If you can remove it and it seems to be paused/unpaused in Redux at the right times, I think we can remove it. If this also means `setPauseQueueIncomingRequest` is no longer used we can remove that

Reply:
Actually, we can probably keep `setPauseQueueIncomingRequest` as Patrick is using it separately in his PR for app offline mode.